nikola/cli.py

```diff
@@ -0,0 +1,30 @@
+import argparse
+
+
+def parse_args(args=None):
+    parser = argparse.ArgumentParser()
+    add_arguments(parser)
+    options, args = parser.parse_known_args(args)
+    options = process_options(options) or options
+    return options, args
+
+
+def add_arguments(parser):
+    parser.add_argument('--conf', dest='conf_filename')
+    parser.add_argument('-D', '--define', dest='defines', action='append')
```

Comment:
Nikola’s settings are often more complex than just strings. This setting can be useful for some cases, although I don’t think we really need it, because of the limitations, and because it’s trivial to just edit `conf.py` and undo if the setting doesn’t work for you.